pandas/core/indexes/base.py

```diff
@@ -4317,6 +4317,8 @@ def _ensure_index(index_like, copy=False):
 
 
 def _get_na_value(dtype):
+    # TODO: remove; this may require a deprecation cycle because this is
+    # listed in indexes.api.__all_
```

Comment:
That's odd that that's the case!  After all, the underscore means it should be private...

Reply:
this is not public and can be removed. the whole pandas.core.index bizness needs to be removed as well.